### DISCUSSION TITLE
Orientation icon is the wrong direction

### DISCUSSION THREAD

**Discussion by kevincox** [photoprism/photoprism#3641] [Q&A] [answered]
#### 1. What is not working as documented?

When changing the orientation of a photo the left and right rotation options appear to be swapped.

[Screencast from 2023-08-16 10-38-56.webm](https://github.com/photoprism/photoprism/assets/494012/c031f1a9-a32e-4a88-9279-f6ab6ac23ac6)

This may be some personal preference or lack of understanding but it seems very intuitive. In the video you can see that the image original has its top facing to the right. So I select the orientation icon that has the top to the right. However the resulting image is now upside down. Instead the "correct" option to select is the icon with the top to the left.

#### 2. How can we reproduce it?

1. Have an image that is rotated 90° in either direction.
2. Click edit
3. Go to the files tab.
4. Change the orientation option.

#### 3. What behavior do you expect?

Selecting the icon that matches the rotation of my image results in the image being upright afterwards.

I see two solutions here:

1. Update the icons and behaviour to "select the current orientation of the picture" and the rotation is adjusted accordingly. This should take the current orientation into account. This is what the current iconography implies to me, but it doesn't work that way.
2. Update the icons to clearly reflect rotation instead of selecting the right orientation.

Note that both of these systems would need to take into account the current orientation when selecting which "target" orientation to apply to the image. Right now this is the responsibility of the user which is relatively complicated mental math which could be offloaded to the software.

#### 4. What could be the cause of your problem?

It seems that the icons are intended as a "rotate" icon, but that isn't reflected in the image they contain.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

No. It happens for any image.

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: PhotoPrism® CE Build 230615-90a18f6e7, x86-64

(b) Database Type & Version: SQLite

(c) Operating System Types & Versions: Linux

(d) Browser Types & Versions: Firefox

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? N/A

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: AMD Ryzen

(b) Physical Memory & Swap Space in GB: N/A

(c) Storage Type: SSD

(d) Anything else that might be helpful to know?

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

Nginx

**Comment by lastzero** [comment]
It could be that in your case the image data does not match the orientation metadata? When I tested the implementation on our public demo, everything looked fine and I wouldn't change a thing - let alone consider it a bug that the dropdown (icon) looks exactly the same as the thumbnail image:

![Screenshot 2023-08-16 at 17 00 19](https://github.com/photoprism/photoprism/assets/301686/e54a7626-43e6-4728-bbf2-b8cb9750ae60)

**Reply by lastzero** [reply]
Note that it is not possible to analyze what the image looks like and adjust the icon accordingly. If it was taken in space, there is not even a clear top and bottom.

**Reply by kevincox** [reply]
This is actually showing the problem though. This issue is that you started with an image that was oriented correctly. So when you select an icon it ends up moving to match the icon. But if your source image is oriented differently (so that up is not up). Then the icon doesn't make sense anymore. Since it would be an odd case to make a picture the wrong way up the icon really doesn't work well.

Maybe it would be better to use an icon that reflects rotation such as GIMP:

![image](https://github.com/photoprism/photoprism/assets/494012/c647b881-c67a-4abc-a1b1-1876f7d3573c)

--

I think it isn't hard to see how it is confusing that if I have an image with the top facing left and I select the icon with the top facing left it is surprising that it doesn't do the right thing.

**Reply by lastzero** [reply]
If we knew the correct orientation right away, you wouldn't need this feature in the first place, right?

**Reply by lastzero** [reply]
I guess that instead of a dropdown that shows the actual orientation (in the metadata), you'd rather have buttons to rotate left/right next to the image? So basically the same functionality, just implemented differently?

**Reply by lastzero** [reply]
It may be helpful to know that this only affects the metadata, not the actual image, like when you rotate it in Photoshop, for example. That's something completely different, even though the effect may be the same for you.

**Reply by kevincox** [reply]
I think having buttons to rotate left and right is probably much easier to understand quickly. Then you don't need to consider 1. the current orientation, 2. the desired orientation then compute which icon is the right one.

Another option that might really helpful is when the user selects the orientation dropdown show the image itself in the four possible orientations. This would make it really easy to select the desired one.

Shittiest mock in the world:

![rotate-mock](https://github.com/photoprism/photoprism/assets/494012/c81f160f-b7e6-4746-adde-b0492ca19f5f)

**Reply by kevincox** [reply]
> If we knew the correct orientation right away, you wouldn't need this feature in the first place, right?

I don't understand this question.

I do know the correct orientation by looking at the image, but it is not obvious how that maps to the orientation options provided. I think the UX could be improved here.

**Reply by lastzero** [reply]
Thanks for your feedback! While we agree that usability is important, PhotoPrism is not an image editor, but focuses on metadata. The dropdown shows the absolute Exif orientation as an icon. Changing that would not serve the purpose, and maintaining the same functionality twice is not possible at this time given all the other feature requests.

**Reply by kevincox** [reply]
I'm not sure I follow. This is about editing the EXIF metadata. It is just providing a convenient UI to do so.

What functionality would need to be implemented twice? This is just proposing using an image to show the option that the user is selecting rather than a confusing icon.

**Reply by lastzero** [reply]
That's what we currently have implemented on a technical level (1, 3, 6, 8): http://sylvana.net/jpegcrop/exif_orientation.html

If you require changes to the user interface or functionality for your personal use, this is not possible at this time due to the many other features we are working on.

**Reply by kevincox** [reply]
I understand that this is not a high priority. But it seems useful to track as an area where some UX polish could be valuable.

**Reply by lastzero** [reply]
Yeah, but you originally opened a bug report, which has the highest priority due to our zero bugs policy :)

**Reply by kevincox** [reply]
Sure, sounds like it should be converted to a feature request/improvement suggestion.

**Reply by graciousgrey** [reply, marked as answer]
Feel free to leave a :+1: in there :)
- https://github.com/photoprism/photoprism/issues/3643
